enterprise/server/backends/pebble_cache/pebble_cache_test.go

```diff
@@ -2165,6 +2167,148 @@ func TestDeleteEmptyDirs(t *testing.T) {
+		assert.Equal(t, int64(filestore.Version2), versionMetadata.MaxVersion)
+
+		pc.Stop()
+	}
+}
+
+func TestSpecifiedActiveKeyVersion_NewDatabase(t *testing.T) {
+	te := testenv.GetTestEnv(t)
+	te.SetAuthenticator(testauth.NewTestAuthenticator(emptyUserMap))
+	ctx := getAnonContext(t, te)
+	rootDir := testfs.MakeTempDir(t)
+	maxSizeBytes := int64(1_000_000) // 1MB
+	options := &pebble_cache.Options{RootDirectory: rootDir, MaxSizeBytes: maxSizeBytes}
+
+	// Confirm that a newly-created pebble database with an specified key
+	// version is written at that key version.
+	activeKeyVersion := int64(filestore.Version2)
+	options.ActiveKeyVersion = &activeKeyVersion
+	pc := openPebbleCache(ctx, t, te, options, []string{"remote-instance-name-1"})
+	versionMetadata, err := pc.DatabaseVersionMetadata()
+	require.NoError(t, err)
+	assert.Equal(t, int64(filestore.Version2), versionMetadata.MinVersion)
+	assert.Equal(t, int64(filestore.Version2), versionMetadata.MaxVersion)
+
+	pc.Stop()
+}
+
+func TestSpecifiedActiveKeyVersion_ExistingDatabase(t *testing.T) {
+	te := testenv.GetTestEnv(t)
+	te.SetAuthenticator(testauth.NewTestAuthenticator(emptyUserMap))
+	ctx := getAnonContext(t, te)
+	rootDir := testfs.MakeTempDir(t)
+	maxSizeBytes := int64(1_000_000) // 1MB
+	options := &pebble_cache.Options{RootDirectory: rootDir, MaxSizeBytes: maxSizeBytes}
+
+	// Create a database with version 2 keys
+	{
+		activeKeyVersion := int64(filestore.Version2)
+		options.ActiveKeyVersion = &activeKeyVersion
+		pc := openPebbleCache(ctx, t, te, options, []string{"remote-instance-name-1"})
+		versionMetadata, err := pc.DatabaseVersionMetadata()
+		require.NoError(t, err)
+		assert.Equal(t, int64(filestore.Version2), versionMetadata.MinVersion)
+		assert.Equal(t, int64(filestore.Version2), versionMetadata.MaxVersion)
+
+		pc.TestingWaitForGC()
+		pc.Stop()
+	}
+
+	// Re-open the database with version 3 as the active key version and
+	// confirm the version metadata is [2, 3]. Don't write anything though, to
+	// avoid migrating the data.
+	{
+		activeKeyVersion := int64(filestore.Version3)
+		options.ActiveKeyVersion = &activeKeyVersion
+		pc := openPebbleCache(ctx, t, te, options, []string{})
+		versionMetadata, err := pc.DatabaseVersionMetadata()
+		require.NoError(t, err)
+		assert.Equal(t, int64(filestore.Version2), versionMetadata.MinVersion)
+		assert.Equal(t, int64(filestore.Version3), versionMetadata.MaxVersion)
+
+		pc.Stop()
+	}
+
+	// Re-open the database with version 1 as the active key version and
+	// confirm the version metadata is [1, 3].
+	{
+		activeKeyVersion := int64(filestore.Version1)
+		options.ActiveKeyVersion = &activeKeyVersion
+		pc := openPebbleCache(ctx, t, te, options, []string{"remote-instance-name-3"})
+		versionMetadata, err := pc.DatabaseVersionMetadata()
+		require.NoError(t, err)
+		assert.Equal(t, int64(filestore.Version1), versionMetadata.MinVersion)
+		assert.Equal(t, int64(filestore.Version3), versionMetadata.MaxVersion)
+
+		pc.Stop()
+	}
```

Comment:
Don't know if it's valuable to add this, but your test cases always expand the version range. What should happen if we pass a version within the range? 

i.e. if version metadata is [1,3] and we set activeKeyVersion=2

Reply:
Sure, sounds good.